### PR TITLE
Added '/api/revenues?timeframe=today'

### PR DESCRIPTION
**Pull request by EzeibekweEmma**
# Fixes Issue/Linear Ticket

> This could be an existing issue or a linear ticket
>
> - Linear Ticket Example: Fixes ID-#{COM-39}

# Changes proposed

> Add 'GET /api/revenues?timeframe=today' route for fetching today's revenue data, with parameter validation and success handling.

# Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: src/routes/revenue.route.ts [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
 
   initializeRoutes() {
     this.router.patch(
       `${this.path}/:order_id`,
-      useCatchErrors(this.revenueController.updateRevenue.bind(this.revenueController)),
+      useCatchErrors(this.revenueController.updateRevenue.bind(this.revenueController))
+    );
+
+    // Route for /api/revenues?timeframe=today
+    this.router.get(
```
Comment (Benrobo) [inline src/routes/revenue.route.ts:22]: why did you change the base `path` when you can easily add s at the end of base path`${this.path}s` ? please revert it back as all other endpoint without an `s` depends on it.

File: src/routes/revenue.route.ts [hunk 2]
```diff
@@ -1,20 +1,28 @@
 import express from 'express';
 import useCatchErrors from '../error/catchErrors';
 import RevenueController from '../controller/revenue.controller';
+import { isAuthenticated } from '../middlewares/auth';
 
 export default class RevenueRoute {
   router = express.Router();
   revenueController = new RevenueController();
   path = '/revenues';
```
Comment (Benrobo) [inline src/routes/revenue.route.ts]: change this back to `revenue`

**Review by Benrobo — CHANGES_REQUESTED** [review]

File: src/routes/revenue.route.ts (hunk 2 above)
Comment (EzeibekweEmma) [reply src/routes/revenue.route.ts]: done

**Review by Benrobo — APPROVED** [review]
ok

**Pull request merged**